src/doc/CreateHighlightDialog.js

```diff
@@ -79,12 +79,18 @@ class CreateHighlightDialog extends EventEmitter {
      * @return {void}
      */
     unmountPopover() {
-        this.isVisible = false;
-        const popoverLayer = this.container.querySelector('.ba-dialog-layer');
-        if (this.createPopoverComponent && popoverLayer) {
-            unmountComponentAtNode(popoverLayer);
-            this.createPopoverComponent = null;
+        if (!this.isVisible) {
+            return;
         }
+
+        const popoverLayers = this.container.querySelectorAll('.ba-dialog-layer');
+        if (!this.createPopoverComponent || popoverLayers.length === 0) {
+            return;
```

Comment:
Would we not want to still set `isVisible` to false in this case? It'll get left as true due to the guard clause above, right?